src/robot/libraries/BuiltIn.py

```diff
@@ -1662,26 +1662,55 @@ def run_keyword_and_expect_error(self, expected_error, name, *args):
         return unic(error)
 
     @run_keyword_variant(resolve=2)
-    def repeat_keyword(self, times, name, *args):
+    def repeat_keyword(self, repeat, name, *args):
         """Executes the specified keyword multiple times.
 
         ``name`` and ``args`` define the keyword that is executed
-        similarly as with `Run Keyword`, and ``times`` specifies how many
-        the keyword should be executed. ``times`` can be given as an
+        similarly as with `Run Keyword`.
+
+        If ``repeat`` is given as count, it specifies how many the keyword
+        should be executed. ``repeat`` can be given as an
         integer or as a string that can be converted to an integer. If it is
         a string, it can have postfix ``times`` or ``x`` (case and space
         insensitive) to make the expression more explicit.
 
-        If ``times`` is zero or negative, the keyword is not executed at
+        If ``retry`` is given as timeout, it must be in Robot Framework's
+        time format (e.g. ``1 minute``, ``2 min 3 s``) except a number like
+        42 is interpreted as count, so that a floating point number like
+        4.2 would raise exception.
+
+        If ``retry`` is zero or negative, the keyword is not executed at
         all. This keyword fails immediately if any of the execution
         rounds fails.
 
         Examples:
         | Repeat Keyword | 5 times | Go to Previous Page |
         | Repeat Keyword | ${var}  | Some Keyword | arg1 | arg2 |
+        | Repeat Keyword | 2 mins  | Some Keyword | arg1 | arg2 |
         """
-        times = self._get_times_to_repeat(times)
-        self._run_keywords(self._yield_repeated_keywords(times, name, args))
+        try:
+            count = self._get_times_to_repeat(repeat, require_postfix=False)
+        except RuntimeError, e:
+            if is_string(repeat) and repeat.replace('.', '', 1).isdigit() or \
+                    is_number(repeat):
+                # raise exception not to translate as time string for
+                # backward compatibility
+                raise e
+            try:
+                timeout = timestr_to_secs(repeat)
+                maxtime = time.time() + timeout
+            except:
+                # if it's failed to convert repeat to time, raise exception
+                # originally occurred by _get_times_to_repeat() for
+                # backward compatibility
+                # Without it, "Invalid Times" in acceptance test fails.
+                raise e
+            else:
+                kwiter = self._yield_repeated_keywords_until(maxtime, name,
+                                                             args)
+        else:
+            kwiter = self._yield_repeated_keywords(count, name, args)
+        self._run_keywords(kwiter)
```

Comment:
The above code is pretty complicated with nested exception handlers etc. and doesn't fully follow our coding guidelines otherwise either. Please try to clean it up. Things to take into account:

- Try to separate handling repeat as count and maxtime more clearly.
- You should be able to write the code so cleanly that it needs no comments.
- The code needs to be Python 3 compatible. The current exception handling syntax is not.
- Avoid one char variables except in tight loops. Use e.g. `err` or `error` instead of `e`.

If you need more help, please first check code conventions found from CONTRIBUTING.rst file. If that's not enough, I can comment more when I have time.